scripts/4.Covariates.py

```diff
@@ -0,0 +1,222 @@
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+get_ipython().magic('matplotlib inline')
+
+
+# Let's peak at the sample data.
+
+# In[2]:
+
+samples_df = pd.read_table('download/samples.tsv', index_col=0)
+samples_df.head(10)
+
+
+# Let's get a sense of how much missing data there is.
+
+# In[3]:
+
+print('Total number of samples: %d' % len(samples_df))
+print('Number of nulls in each column:')
+samples_df.isnull().sum(axis=0)
+
+
+# Now count the number of types of each column. We'll use Python's set, and therefore must exclude null values.
+
+# In[4]:
+
+categorical_variables = samples_df.columns[[1, 2, 3, 4, 6, 8]]
+for variable in categorical_variables:
+    not_null = ~samples_df[variable].isnull()
+    count = len(set(samples_df.loc[not_null, variable]))
+    print('Number of types in category %s: %d' % (variable, count))
+
+
+# There are no missing values from <code>sample_types</code> and only one possible value, so it is redundant and we'll exclude it.
+
+# In[5]:
+
+categorical_variables = categorical_variables[1:]
+print(categorical_variables)
+
+
+# Inspecting the head of the samples DataFrame above, we see that two columns, namely <code>dead</code> and <code>recurred</code>, need some attention. We're going to encode categorical variables using panda's get_dummies. Since these columns are indicated by a 1 or 0, this will become the column header when encoded, as below.
+
+# In[6]:
+
+pd.get_dummies(samples_df.dead).head(10)
+
+
+# Note that null values are encoded as (0,0) pairs... I was wondering if we were okay with this.
+# 
+# Let's rename the values in each of these so that they more accurately reflect the underlying variable.
+
+# In[7]:
+
+mortality_map = pd.Series(['alive', 'dead'], index=[0.0, 1.0])
+mortality = samples_df.dead.map(mortality_map)
+mortality.head(10)
+
+
+# In[8]:
+
+recurred_map = pd.Series(['has_not_recurred', 'has_recurred'], index=[0.0, 1.0])
+recurred = samples_df.recurred.map(recurred_map)
+recurred.head(10)
+
+
+# Now when we concatenate, the columns are appropriately named, and the values are encoded as numeric data that can be input to the types of classifiers that we have been using.
+
+# In[9]:
+
+pd.concat([pd.get_dummies(mortality), pd.get_dummies(recurred)], axis=1).head(10)
+
+
+# Let's process the entire thing. We'll convert the categorical variables to dummies, and then concatenate the numeric variables (age of diagnosis, days survived, and days recurrence free).
+
+# In[10]:
+
+columns_to_process = [samples_df.disease,
+                      samples_df.organ_of_origin,
+                      samples_df.gender,
+                      mortality,
+                      recurred]
+columns_as_dummies = [pd.get_dummies(column) for column in columns_to_process]
```

Comment:
This ties into your comment above about using a big dictionary to rename all the columns. When you use `get_dummies` on the whole data frame, it prepends the name of the column to each categorical variable. I.e., the new columns names for the `recurred` column will be `recurred_has_recurred` and `recurred_has_not_recurred`.

Are we okay with this? Without also formatting the disease/organ names, it starts to look a little strange. For instance, `organ_of_origin_Thyroid Gland` as a column name. Typing out the big dictionary is doable but slightly tedious... perhaps there is room to be clever with Python or regexes.

It's definitely minor, but any advice on how to proceed for the next PR would be appreciated.